### ISSUE TITLE
[new rule] Don't allow toMatchObject with objectContaining

### ISSUE THREAD

**Issue by geoffswift** [jest-community/eslint-plugin-jest#1734]
I observed some examples like this in my code base, and it's redundant.  So a new rule to eliminate this would be nice.

FAIL:
```javascript
expect(actual).toMatchObject(
  expect.objectContaining(expected)
);
```

WIN:
```javascript
expect(actual).toMatchObject(expected);
```

**Comment by G-Rath** [comment]
fwiw ironically the unofficial recommendation is actually to use `toEqual(expect.objectContaining)`, rather than `toMatchObject` and @SimenB has mentioned in the past wanting to ideally remove the latter all together: https://github.com/jest-community/eslint-plugin-jest/issues/230

(I say "unofficial" because we don't have an explicit lint rule or notice or anything that actually tells you this if you use `toMatchObject`, so it's currently up to you to decide if you do want to use it  or not)

**Comment by geoffswift** [comment]
Thanks for clarifying.  I read the thread cited, and it lead me here https://stackoverflow.com/questions/45692456/whats-the-difference-between-tomatchobject-and-objectcontaining

I can see what I'm suggesting wouldn't lead to equivalent code and therefore doesn't make for a good rule!
